### ISSUE TITLE
Pass function to delete row in the column definition

### ISSUE THREAD

**Issue by ilbassa** [shadcn-ui/ui#2017]
I have a function for deleting a row

`const { mutate: deleteRow } = trpc.site.delete.useMutation()`

in the same file i render the DataTabel component passing the function
`<DataTable
              columns={columns}
              data={data}
              className="bg-white"
              handleDeleteRow={deleteRow}
            />`

How I can use deleteRow function in the dropdown menù?

```{
    id: "actions",
    cell: ({ row }: any) => {
      const site = row.original;
      return (
        <div className="flex gap-2">
          <Link href={`/admin/site/${site.id}`}>
            <Button
              className={buttonVariants({
                size: "sm",
                className: "my-4",
                variant: "default",
              })}
            >
              View
            </Button>
          </Link>{" "}
         <Button
            onClick={() => deleteRow({ id: site.id })}
            size="sm"
            className="my-4"
            variant="destructive"
          >
           Delete
            </Button> 
        </div>
      );
    },
  },```

**Comment by ilbassa** [comment]
No... I'm still stuck.
If you find something please update this issue 😊

**Comment by vikrant0526** [comment]
You'll need to hook on to the onSuccess of the mutation (deleteRow) to remove that row from the original data.

Here's an example
`
// use this to hold your table data and modify it per your choice.
const { mutation:deleteRow } = useMutation( {
  ...,
  onSuccess: (result) => { setData( state => state.filter( site => site.id !== result.id) ),
} )
`

**OR**

`
const { mutation:deleteRow } = useMutation( {...} )

return <Button
  onClick={() => deleteRow({ id: site.id }, {
  ...,
  onSuccess: (result) => { setData( state => state.filter( site => site.id !== result.id) ),
  })}
  >
    Delete
 </Button> 
`

Basically you'll need to remove the row from the table data once the mutation has succeeded.
Here's the [docs](https://tanstack.com/query/v3/docs/react/guides/mutations) from react-query (hooks provided by @trpc/react are a thin wrapper around React Query) which should be helpful.

**Comment by vikrant0526** [comment]
Just for future reference this question involves trpc, react-query and react-table repositories and has nothing to do with shadcn/ui and StackOverflow would be the more appropriate place for questions like this.

**Comment by ilbassa** [comment]
@vikrant0526 thank you for your response.
I need ti pass the function to the drop down menù tò use it in the onClick function of a button.
The problem is: how to pass the function to the drop down?

**Comment by vikrant0526** [comment]
> @vikrant0526 thank you for your response. I need ti pass the function to the drop down menù tò use it in the onClick function of a button. The problem is: how to pass the function to the drop down?

In that case you can modify your data rows directly to add that function.

```typescript
<Table data={data.map((site => return {...site, deleteRow: ()=>deleteRow({ id: site.id }))} />
```

and in your column definition

```typescript
cell: ({ row }: any) => {
      const site = row.original;
      return (
        <div className="flex gap-2">
          <Link href={`/admin/site/${site.id}`}>
            <Button
              className={buttonVariants({
                size: "sm",
                className: "my-4",
                variant: "default",
              })}
            >
              View
            </Button>
          </Link>{" "}
         <Button
            onClick={deleteRow} // or onClick={()=>deleteRow()}
            size="sm"
            className="my-4"
            variant="destructive"
          >
           Delete
            </Button> 
        </div>
      );
    },
```

**Comment by shadcn** [comment]
This issue has been automatically closed because it received no activity for a while. If you think it was closed by accident, please leave a comment. Thank you.
